modules/src/main/java/org/jboss/additional/testsuite/jdkall/present/ejb/lookup/LookupToInvalidHostNameTestCase.java

```diff
@@ -0,0 +1,115 @@
+import java.util.Properties;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+
+/**
+ * @author Jiri Bilek
+ * jbilek@redhat.com on 12/03/18.
+ * Test for XNIO-320
+ *
+ * IllegalArgumentException: Parameter 'address' may not be null
+ * if invalid-hostname does not resolve to an IP, then you will see the error.
+ *
+ * Correct behavior should return
+ * javax.naming.CommunicationException: WFNAM00018: Failed to connect to remote host [Root exception is java.net.UnknownHostException: invalid-hostname]
+ * ...
+ * Caused by: java.net.UnknownHostException: invalid-hostname
+ *
+ * Incorrect behavior returns
+ * java.lang.IllegalArgumentException: Parameter 'address' may not be null
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@EapAdditionalTestsuite({"modules/testcases/jdkAll/Wildfly/ejb/src/main/java","modules/testcases/jdkAll/Eap7/ejb/src/main/java","modules/testcases/jdkAll/Eap71x-Proposed/ejb/src/main/java","modules/testcases/jdkAll/Eap71x/ejb/src/main/java"})
+public class LookupToInvalidHostNameTestCase {
+
+   private static final String ARCHIVE_NAME = "test-simple-ejb";
+   private static final String DEPLOYMENT = ARCHIVE_NAME + ".jar";
+
+   @Deployment(name = DEPLOYMENT)
+   public static Archive<?> deploy() {
+      JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
+      jar.addPackage(TestRemote.class.getPackage());
+      jar.addPackage(TestSLSB.class.getPackage());
+      return jar;
+   }
+
+   /*
+   get inicial context and lookup
+    */
+   public TestRemote lookup(String host) throws Exception {
+      Properties props = new Properties();
+      props.put(Context.INITIAL_CONTEXT_FACTORY,  "org.wildfly.naming.client.WildFlyInitialContextFactory");
+      props.put(Context.PROVIDER_URL, String.format("remote+http://%s:%d", host, 8080));
+      Context ic = new InitialContext(props);
+      // from some reason the deployment does not have name DEPLOYMENT but "test.war"
+      final String lookup = "test//" + TestSLSB.class.getSimpleName() + "!" + TestRemote.class.getName();
+      return (TestRemote) ic.lookup(lookup);
+   }
+
+   /*
+   test if ejb is correctly deployed and accessible with using correct hostname
+    */
+   @Test
+   @OperateOnDeployment(DEPLOYMENT)
+   public void testLookUpWithCorrectHostname() throws Exception {
+      TestRemote proxy = lookup(TestSuiteEnvironment.getServerAddress());
+      String echo = "Answer to the Ultimate Question of Life, the Universe, and Everything";
+      String result = proxy.invoke(echo);
+      Assert.assertEquals(echo,result);
+   }
+
+   /*
+   check if correct exception is returned in case invalid hostname
+    */
+   @Test(expected = CommunicationException.class)
+   @OperateOnDeployment(DEPLOYMENT)
+   public void testLookUpWithBadHostname() throws Exception {
+      TestRemote proxy = lookup("invalid-hostname");
+      String echo = "Answer to the Ultimate Question of Life, the Universe, and Everything";
+      proxy.invoke(echo);
```

Comment:
This will maybe cause the test to fail on the CI

Reply:
Why you think that? The test expects CommunicationException so when behavior is correct the test should be passed. Incorrect behavior invokes IllegalArgumentException.
It is failing now probably because the fix is included in xnio taf which is not in any bom yet.

Reply:
you mean line 107, ok